python/lsst/ip/diffim/getTemplate.py

```diff
@@ -644,31 +634,32 @@ def run(self, coaddExposures, bbox, wcs):
         maskedImageList = []
         weightList = []
 
-        for coaddExposure in coaddExposures:
-            coaddPatch = coaddExposure.get()
+        for coaddExposure, dataId in zip(coaddExposures, dataIds):
 
             # warp to detector WCS
-            warped = self.warper.warpExposure(finalWcs, coaddPatch, maxBBox=finalBBox)
+            warped = self.warper.warpExposure(finalWcs, coaddExposure, maxBBox=finalBBox)
 
             # Check if warped image is viable
             if not np.any(np.isfinite(warped.image.array)):
-                self.log.info("No overlap for warped %s. Skipping" % coaddExposure.ref.dataId)
+                self.log.info("No overlap for warped %s. Skipping" % dataId)
                 continue
 
             exp = afwImage.ExposureF(finalBBox, finalWcs)
             exp.maskedImage.set(np.nan, afwImage.Mask.getPlaneBitMask("NO_DATA"), np.nan)
-            exp.maskedImage.assign(warped.maskedImage, warped.getBBox())
+            warpedBBox = warped.getBBox()
+            warpedBBox.clip(finalBBox)
```

Comment:
These 2 added lines  (and mod one line under) are not necessary: `warpExposure` clips the destination BBox to the `maxBBox=finalBBox`.